File: post_to_slack.R
library(httr)
library(jsonlite)

source("fun.R")
source("credentials.R")

papers <- read_json("./output/papers.json", simplifyVector=TRUE)

# Post Top 10 to Slack 
papers <- papers[papers$score>=0.5,]
papers <- subset(papers, specialized==FALSE)

all_papers_to_slack(subset(papers, preprint==FALSE))
all_papers_to_slack(subset(papers, preprint==TRUE))

File: post_to_slack.py
import json
import pandas as pd

from fun import all_papers_to_slack

with open("./output/papers.json", "r") as f:
    papers = pd.DataFrame(json.load(f))

# Post to Slack
papers = papers[(papers["score"] >= 0.5) & ~papers["specialized"]]

all_papers_to_slack(papers[~papers["preprint"]])
all_papers_to_slack(papers[papers["preprint"]])
